linkerd/app/core/src/errors/respond.rs

```diff
@@ -251,13 +141,80 @@ impl<RspB: Default + hyper::body::HttpBody> respond::Respond<http::Response<RspB
     }
 }
 
+// === impl ResponseBody ===
+
+impl<B: hyper::body::HttpBody> hyper::body::HttpBody for ResponseBody<B>
+where
+    B::Error: Into<Error>,
+{
+    type Data = B::Data;
+    type Error = B::Error;
+
+    fn poll_data(
+        self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Option<Result<Self::Data, Self::Error>>> {
+        match self.project() {
+            ResponseBodyProj::NonGrpc(inner) => inner.poll_data(cx),
+            ResponseBodyProj::Grpc { inner, trailers } => {
+                // should not be calling poll_data if we have set trailers derived from an error
+                assert!(trailers.is_none());
```

Comment:
nit/TIOLI: should we maybe just move the comment into the panic message for the assert?